packages/components/resource-pages/list-page.tsx

```diff
@@ -0,0 +1,246 @@
+        id: tableColumnInfo[2].id
+      },
+      {
+        title: t("Buckets"),
+        props: {
+          className: tableColumnInfo[3].className
+        },
+        id: tableColumnInfo[3].id
+      },
+      {
+        title: "",
+        props: {
+          className: tableColumnInfo[4].className
+        },
+        id: tableColumnInfo[4].id
+      }
+    ],
+    [t]
+  );
+
+  const [columns] = useActiveColumns({
+    columns: tableColumns,
+    showNamespaceOverride: false,
+    columnManagementID: null
+  });
+  return (
+    <VirtualizedTable
+      {...props}
+      aria-label={t("Resource Page")}
+      columns={columns}
+      Row={RowRenderer}
+    />
+  );
+};
+
+type CustomData = {
+  launchModal: LaunchModal;
+  resourceModel: K8sModel;
+  kebabActions?: (t: TFunction) => { [key: string]: { value: string } };
+};
+
+const RowRenderer: React.FC<RowProps<K8sResourceCommon, CustomData>> = ({
+  obj,
+  activeColumnIDs,
+  rowData
+}) => {
+  const { resourceModel, launchModal, kebabActions } = rowData;
+  const name = getName(obj);
+  const provider = getProvider(obj);
+  return (
+    <>
+      <TableData {...tableColumnInfo[0]} activeColumnIDs={activeColumnIDs}>
+        {name}
+      </TableData>
+      <TableData {...tableColumnInfo[1]} activeColumnIDs={activeColumnIDs}>
+        <OperandStatus operand={obj} />
+      </TableData>
+      <TableData {...tableColumnInfo[2]} activeColumnIDs={activeColumnIDs}>
+        {provider}
+      </TableData>
+      <TableData {...tableColumnInfo[3]} activeColumnIDs={activeColumnIDs}>
+        {2 + " Buckets"}
+      </TableData>
+      <TableData {...tableColumnInfo[4]} activeColumnIDs={activeColumnIDs}>
+        <Kebab
+          launchModal={launchModal}
+          extraProps={{ resource: obj, resourceModel: resourceModel }}
+          customKebabItems={kebabActions}
+        />
+      </TableData>
+    </>
+  );
+};
+
+type GenericListPageProps = {
+  actions?: ModalMap;
+  resourceModel: K8sModel;
+  kebabActions?: CustomData["kebabActions"];
+};
+
+const GenericListPage: React.FC<GenericListPageProps> = ({
+  actions,
+  resourceModel,
+  kebabActions
+}) => {
+  const { t } = useTranslation();
+  const [Modal, modalProps, launchModal] = useModalLauncher(actions);
+
+  const resource = React.useMemo(
+    () => ({
+      kind: referenceForModel(resourceModel),
+      ns: DATA_FEDERATION_NAMESPACE,
+      isList: true
+    }),
+    [resourceModel]
+  );
+  const [resources, loaded, loadError] =
+    useK8sWatchResource<K8sResourceCommon[]>(resource);
+  const [data, filteredData, onFilterChange] = useListPageFilter(resources);
+  const new_filtered_array = (filteredData as NamespaceStoreKind[])
+    .filter((a) => {
+      return a.metadata.namespace === DATA_FEDERATION_NAMESPACE;
+    })
+    .map((a) => {
+      return {
+        ...a,
+        provider: Object.keys(NOOBAA_TYPE_MAP).find(
+          (key) => NOOBAA_TYPE_MAP[key] === a.spec.type
+        )
+      };
+    });
```

Comment:
1) Don't filter it here, pass a filter function to `useListPageFilter` if needed.
2) But, we need to check why we have to filter in first place, when we are passing namespace on line 179 then there should be no need to filter, API should return NSS in that particular namespace only. If that is not happening we need to check why that is happening.